<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="tkik" />

<meta name="date" content="2023-06-07" />

<title>GRanges</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ROmics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkik/ROmics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GRanges</h1>
<h4 class="author">tkik</h4>
<h4 class="date">2023-06-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ROmics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0.4). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkikROmicstree6583b3571cb24546ccd03828d2ab8f258e89d1c4targetblank6583b35a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkik/ROmics/tree/6583b3571cb24546ccd03828d2ab8f258e89d1c4" target="_blank">6583b35</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkikROmicstree6583b3571cb24546ccd03828d2ab8f258e89d1c4targetblank6583b35a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkik/ROmics/tree/6583b3571cb24546ccd03828d2ab8f258e89d1c4" target="_blank">6583b35</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Course_plans.docx
    Untracked:  data/idat/
    Untracked:  data/mm9_Refseq_chr12.tsv
    Untracked:  data/normalized_rnb.set.RDS
    Untracked:  data/raw_rnb.set.RDS
    Untracked:  data/raw_rnbset_GSE175758.RDS
    Untracked:  data/sample_annotation.csv
    Untracked:  output/Genomic data.pptx
    Untracked:  output/diffmeth/

Unstaged changes:
    Modified:   analysis/reproducible_programming.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GRanges.Rmd</code>) and HTML
(<code>docs/GRanges.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkik/ROmics/blob/6583b3571cb24546ccd03828d2ab8f258e89d1c4/analysis/GRanges.Rmd" target="_blank">6583b35</a>
</td>
<td>
tkik
</td>
<td>
2023-06-22
</td>
<td>
wflow_publish(c("analysis/methylation_plots.Rmd",
"analysis/GRanges.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkik/ROmics/c2dbce973694b0bd7a6234733e0fd31814d289f8/docs/GRanges.html" target="_blank">c2dbce9</a>
</td>
<td>
tkik
</td>
<td>
2023-06-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkik/ROmics/blob/e357ee1375278957f1abedb03e94a79f92507298/analysis/GRanges.Rmd" target="_blank">e357ee1</a>
</td>
<td>
tkik
</td>
<td>
2023-06-08
</td>
<td>
additional files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="granges-and-grangeslist" class="section level2">
<h2>Granges and GRangesList</h2>
<p><code>GRanges</code> objects store information about Genomic ranges
with chromosome, position and additional information. Offers a wide
range of functionality. <code>GRangesList</code> is a “grouped” version
of <code>GRnages</code>, it is similar to a normal list, but with
additional functionality. Both of them are able to handle strand
information as well.</p>
<pre class="r"><code>library(GenomicRanges)</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:grDevices&#39;:

    windows</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>gr1 &lt;- GRanges(
    seqnames = &quot;chr2&quot;,
    ranges = IRanges(103, 106),
    strand = &quot;+&quot;,
    score = 5L, GC = 0.45)
gr2 &lt;- GRanges(
    seqnames = c(&quot;chr1&quot;, &quot;chr3&quot;),
    ranges = IRanges(c(107, 113), width = 3),
    strand = c(&quot;+&quot;, &quot;-&quot;),
    score = 3:4, GC = c(0.3, 0.5))
gr3 &lt;- GRanges(&quot;chr2:102-107&quot;)

gr4 &lt;- makeGRangesFromDataFrame(data.frame(chr=c(&quot;chr1&quot;, &quot;chr2&quot;), 
                                           start=c(104, 104), end=c(108, 105),
                                           name=c(&quot;gene1&quot;, &quot;gene2&quot;)), keep.extra.columns = T, ignore.strand = T)
grl &lt;- GRangesList(&quot;txA&quot; = gr1, &quot;txB&quot; = gr2)
grl</code></pre>
<pre><code>GRangesList object of length 2:
$txA
GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr2   103-106      + |         5      0.45
  -------
  seqinfo: 3 sequences from an unspecified genome; no seqlengths

$txB
GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr1   107-109      + |         3       0.3
  [2]     chr3   113-115      - |         4       0.5
  -------
  seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>#seqinfo(gr1) &lt;- Seqinfo(genome=&quot;hg38&quot;)
#can have names for each row</code></pre>
<div id="accessors" class="section level3">
<h3>Accessors</h3>
<pre class="r"><code>seqnames(gr1)</code></pre>
<pre><code>factor-Rle of length 1 with 1 run
  Lengths:    1
  Values : chr2
Levels(1): chr2</code></pre>
<pre class="r"><code>seqlevels(gr1)</code></pre>
<pre><code>[1] &quot;chr2&quot;</code></pre>
<pre class="r"><code>ranges(gr1)</code></pre>
<pre><code>IRanges object with 1 range and 0 metadata columns:
          start       end     width
      &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
  [1]       103       106         4</code></pre>
<pre class="r"><code>start(gr1)</code></pre>
<pre><code>[1] 103</code></pre>
<pre class="r"><code>end(gr1)</code></pre>
<pre><code>[1] 106</code></pre>
<pre class="r"><code>strand(gr1)</code></pre>
<pre><code>factor-Rle of length 1 with 1 run
  Lengths: 1
  Values : +
Levels(3): + - *</code></pre>
<pre class="r"><code>width(gr1)</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code>length(gr1)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>mcols(gr4)</code></pre>
<pre><code>DataFrame with 2 rows and 1 column
         name
  &lt;character&gt;
1       gene1
2       gene2</code></pre>
<pre class="r"><code>mcols(gr4)$name</code></pre>
<pre><code>[1] &quot;gene1&quot; &quot;gene2&quot;</code></pre>
</div>
<div id="subsetting" class="section level3">
<h3>Subsetting</h3>
<pre class="r"><code>#data.frame-like

gr1[1,]</code></pre>
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr2   103-106      + |         5      0.45
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>gr4[1, &quot;name&quot;]</code></pre>
<pre><code>GRanges object with 1 range and 1 metadata column:
      seqnames    ranges strand |        name
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr1   104-108      * |       gene1
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code># subset function
subset(gr1, strand == &quot;+&quot;)</code></pre>
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr2   103-106      + |         5      0.45
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="splitting-and-combining" class="section level3">
<h3>Splitting and combining</h3>
<pre class="r"><code>sp &lt;- split(gr2, 1:2)
split(gr4, ~ name)</code></pre>
<pre><code>GRangesList object of length 2:
$gene1
GRanges object with 1 range and 1 metadata column:
      seqnames    ranges strand |        name
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr1   104-108      * |       gene1
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths

$gene2
GRanges object with 1 range and 1 metadata column:
      seqnames    ranges strand |        name
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr2   104-105      * |       gene2
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>c(sp[[1]], sp[[2]])</code></pre>
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr1   107-109      + |         3       0.3
  [2]     chr3   113-115      - |         4       0.5
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>stack(sp, index.var=&quot;name&quot;)</code></pre>
<pre><code>GRanges object with 2 ranges and 3 metadata columns:
      seqnames    ranges strand |  name     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;Rle&gt; &lt;integer&gt; &lt;numeric&gt;
  [1]     chr1   107-109      + |     1         3       0.3
  [2]     chr3   113-115      - |     2         4       0.5
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="aggregating" class="section level3">
<h3>Aggregating</h3>
<pre class="r"><code>aggregate(gr2, score ~ strand, mean)</code></pre>
<pre><code>DataFrame with 2 rows and 2 columns
    strand     score
  &lt;factor&gt; &lt;numeric&gt;
1        +         3
2        -         4</code></pre>
</div>
<div id="interval-operations" class="section level3">
<h3>Interval operations</h3>
<pre class="r"><code>flank(gr1, 10, both=T)</code></pre>
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr2    93-112      + |         5      0.45
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>shift(gr2, 5)</code></pre>
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr1   112-114      + |         3       0.3
  [2]     chr3   118-120      - |         4       0.5
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>resize(gr1, 20)</code></pre>
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr2   103-122      + |         5      0.45
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>c(gr1, gr3)</code></pre>
<pre><code>GRanges object with 2 ranges and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr2   103-106      + |         5      0.45
  [2]     chr2   102-107      * |      &lt;NA&gt;        NA
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code> reduce(c(gr1, gr3)) # combines the overlapping ranges</code></pre>
<pre><code>GRanges object with 2 ranges and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr2   103-106      +
  [2]     chr2   102-107      *
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code> reduce(c(gr1, gr3), ignore.strand=T)</code></pre>
<pre><code>GRanges object with 1 range and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr2   102-107      *
  -------
  seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="comapring-2-granges" class="section level3">
<h3>Comapring 2 GRanges</h3>
<pre class="r"><code>union(gr2, gr4)</code></pre>
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): Each of the 2 combined objects has sequence levels not in the other:
  - in &#39;x&#39;: chr3
  - in &#39;y&#39;: chr2
  Make sure to always combine/compare objects based on the same reference
  genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>GRanges object with 4 ranges and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1   107-109      +
  [2]     chr1   104-108      *
  [3]     chr3   113-115      -
  [4]     chr2   104-105      *
  -------
  seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>intersect(gr2, gr4)</code></pre>
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): Each of the 2 combined objects has sequence levels not in the other:
  - in &#39;x&#39;: chr3
  - in &#39;y&#39;: chr2
  Make sure to always combine/compare objects based on the same reference
  genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>GRanges object with 0 ranges and 0 metadata columns:
   seqnames    ranges strand
      &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  -------
  seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>setdiff(gr2, gr4)</code></pre>
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): Each of the 2 combined objects has sequence levels not in the other:
  - in &#39;x&#39;: chr3
  - in &#39;y&#39;: chr2
  Make sure to always combine/compare objects based on the same reference
  genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>GRanges object with 2 ranges and 0 metadata columns:
      seqnames    ranges strand
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
  [1]     chr1   107-109      +
  [2]     chr3   113-115      -
  -------
  seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>hits &lt;- findOverlaps(gr2, gr4, ignore.strand=TRUE)</code></pre>
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): Each of the 2 combined objects has sequence levels not in the other:
  - in &#39;x&#39;: chr3
  - in &#39;y&#39;: chr2
  Make sure to always combine/compare objects based on the same reference
  genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre class="r"><code>countOverlaps(gr2, gr4, ignore.strand=TRUE)</code></pre>
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): Each of the 2 combined objects has sequence levels not in the other:
  - in &#39;x&#39;: chr3
  - in &#39;y&#39;: chr2
  Make sure to always combine/compare objects based on the same reference
  genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>[1] 1 0</code></pre>
<pre class="r"><code>subsetByOverlaps(gr2, gr4)</code></pre>
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): Each of the 2 combined objects has sequence levels not in the other:
  - in &#39;x&#39;: chr3
  - in &#39;y&#39;: chr2
  Make sure to always combine/compare objects based on the same reference
  genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>GRanges object with 1 range and 2 metadata columns:
      seqnames    ranges strand |     score        GC
         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
  [1]     chr1   107-109      + |         3       0.3
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Create a GRanges object using the information in the table below: chr
start end strand score chr1 10000 10300 + 10 chr1 11100 11500 - 20 chr2
20000 20030 + 15</p>
<pre class="r"><code>gr1 &lt;- GRanges(c(&quot;chr1:10000-10300:+&quot;, &quot;chr1:11100-11500:-&quot;, &quot;chr2:20000-20030:+&quot;))
mcols(gr1)$score &lt;- c(&quot;10&quot;, &quot;20&quot;, &quot;15&quot;)

start(gr1)</code></pre>
<pre><code>[1] 10000 11100 20000</code></pre>
<pre class="r"><code>end(gr1)</code></pre>
<pre><code>[1] 10300 11500 20030</code></pre>
<pre class="r"><code>strand(gr1)</code></pre>
<pre><code>factor-Rle of length 3 with 3 runs
  Lengths: 1 1 1
  Values : + - +
Levels(3): + - *</code></pre>
<pre class="r"><code>seqnames(gr1)</code></pre>
<pre><code>factor-Rle of length 3 with 2 runs
  Lengths:    2    1
  Values : chr1 chr2
Levels(2): chr1 chr2</code></pre>
<p>Use the start(), end(), strand(),seqnames() and width() functions on
the GRanges object you created. 1. Subset only the ones on the + strand.
2. subset the ones on the chr1</p>
<pre class="r"><code>gr1[strand(gr1)==&quot;+&quot;,]</code></pre>
<pre><code>GRanges object with 2 ranges and 1 metadata column:
      seqnames      ranges strand |       score
         &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr1 10000-10300      + |          10
  [2]     chr2 20000-20030      + |          15
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>gr1[seqnames(gr1)==&quot;chr1&quot;]</code></pre>
<pre><code>GRanges object with 2 ranges and 1 metadata column:
      seqnames      ranges strand |       score
         &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr1 10000-10300      + |          10
  [2]     chr1 11100-11500      - |          20
  -------
  seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Import mouse (mm9 assembly) CpG islands and RefSeq transcripts for
chr12 from the UCSC browser as GRanges objects using rtracklayer
functions. HINT: Check chapter content and modify the code there as
necessary. If that somehow does not work, go to the UCSC browser and
download it as a BED file. The track name for Refseq genes is “RefSeq
Genes” and the table name is “refGene”.</p>
<pre class="r"><code>library(rtracklayer)
library(AnnotationHub)

ahub &lt;- AnnotationHub()
table(ahub$rdataclass)
ahub &lt;- subset(ahub, rdataclass == &quot;GRanges&quot; &amp; species == &quot;Mus musculus&quot; )
#query(ahub, c(&quot;mm9&quot;))
gr_data&lt;- ahub[ahub$title %in% c(&quot;CpG Islands&quot;, &quot;RefSeq Genes&quot;) &amp; ahub$genome==&quot;mm9&quot;]
cpg &lt;- gr_data
#refseq &lt;- gr_data[[1]]

session &lt;- browserSession(&quot;UCSC&quot;,url = &#39;http://genome-euro.ucsc.edu/cgi-bin/&#39;)
genome(session) &lt;- &quot;mm9&quot;
#query &lt;- ucscTableQuery(session, track=&quot;CpG Islands&quot;,table=&quot;cpgIslandExt&quot;,
#        range=GRangesForUCSCGenome(&quot;mm9&quot;, &quot;chr12&quot;))
#cpg &lt;- track(query)
query &lt;- ucscTableQuery(session, track=&quot;RefSeq Genes&quot;,table=&quot;refGene&quot;,
        range=GRangesForUCSCGenome(&quot;mm9&quot;, &quot;chr12&quot;))
refseq &lt;- getTable(query)
refseq &lt;- makeGRangesFromDataFrame(refseq, seqnames.field = &quot;chrom&quot;, start.field = &quot;txstart&quot;, end.field = &quot;txend&quot;, keep.extra.columns = T)

cpg &lt;- cpg[seqnames(cpg)==&quot;chr12&quot;]
refseq &lt;- read.table(&quot;data/mm9_Refseq_chr12.tsv&quot;)

prom &lt;- promoters(refseq, upstream = 1000, downstream = 1000)

percentage &lt;- sum(countOverlaps(cpg, prom)!=0)/length(cpg)</code></pre>
<p>Following from the exercise above, get the promoters of Refseq
transcripts (-1000bp and +1000 bp of the TSS) and calculate what
percentage of them overlap with CpG islands. HINT: You have to get the
promoter coordinates and use the findOverlaps() or subsetByOverlaps()
from the GenomicRanges package. To get promoters, type ?promoters on the
R console and see how to use that function to get promoters or calculate
their coordinates as shown in the chapter.</p>
<p>Find the length of CpG islands that overlap with the promoters.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] GenomicRanges_1.50.2 GenomeInfoDb_1.34.9  IRanges_2.32.0      
[4] S4Vectors_0.36.1     BiocGenerics_0.44.0  workflowr_1.7.0.4   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9             XVector_0.38.0         compiler_4.2.3        
 [4] pillar_1.9.0           bslib_0.5.0            later_1.3.0           
 [7] git2r_0.30.1           jquerylib_0.1.4        zlibbioc_1.44.0       
[10] bitops_1.0-7           tools_4.2.3            getPass_0.2-2         
[13] digest_0.6.29          jsonlite_1.8.5         evaluate_0.21         
[16] lifecycle_1.0.3        tibble_3.1.8           pkgconfig_2.0.3       
[19] rlang_1.1.1            cli_3.6.0              rstudioapi_0.14       
[22] yaml_2.3.6             xfun_0.39              fastmap_1.1.0         
[25] GenomeInfoDbData_1.2.9 httr_1.4.6             stringr_1.5.0         
[28] knitr_1.43             fs_1.5.2               vctrs_0.5.2           
[31] sass_0.4.6             rprojroot_2.0.3        glue_1.6.2            
[34] R6_2.5.1               processx_3.7.0         fansi_1.0.4           
[37] rmarkdown_2.22         callr_3.7.3            magrittr_2.0.3        
[40] whisker_0.4.1          ps_1.7.0               promises_1.2.0.1      
[43] htmltools_0.5.4        httpuv_1.6.6           utf8_1.2.3            
[46] stringi_1.7.8          RCurl_1.98-1.10        cachem_1.0.6          </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
