---
title: "Working with mutation data"
author: "Reka_Toth"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output:
 # html_notebook: default
  workflowr::wflow_html:
    toc: false
    code_folding: "show"
#output: workflowr::wflow_html:
editor_options:
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Working with mutation data 


```{r, eval=F}
cohortMAF <- read.delim("P:/Teaching/ROmics/data/cohortMAF.LIHC")
cohortMAF$Tumor_Sample_Barcode <- substr(cohortMAF$Tumor_Sample_Barcode,1,12)
write.table(cohortMAF, "P:/Teaching/ROmics/data/cohortMAF.LIHC.maf", sep="\t", row.names = F, quote = F)

```

```{r}
library('NMF')
library("BSgenome.Hsapiens.UCSC.hg38", quietly = TRUE)
library(maftools)
lihc_cohort <- maftools::read.maf(maf = "data/cohortMAF.LIHC.maf", clinicalData = "data/clinical.tsv")
maftools::oncoplot(lihc_cohort)

#  #Shows sample summry.
 getSampleSummary(lihc_cohort)
#  #Shows gene summary.
  getGeneSummary(lihc_cohort)
#  #shows clinical data associated with samples
  getClinicalData(lihc_cohort)
#  #Shows all fields in MAF
#  getFields(laml)


```



```{r}


plotmafSummary(maf = lihc_cohort, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)

#oncoplot for top ten mutated genes.
oncoplot(maf = lihc_cohort, top = 10)

lihc_cohort.titv = titv(maf = lihc_cohort, plot = FALSE, useSyn = TRUE)
#plot titv summary
plotTiTv(res = lihc_cohort.titv)

#lollipop plot for DNMT3A, which is one of the most frequent mutated gene in Leukemia.
lollipopPlot(
  maf = lihc_cohort,
  gene = 'TP53',
  AACol = 'HGVSp_Short',
  showMutationRate = TRUE,
)

```


```{r}


lihc.mutload = tcgaCompare(maf = lihc_cohort, cohortName = 'Example-LIHC', logscale = TRUE, capture_size = 50)


lihc_cohort@data$varian_allele_freq <- (lihc_cohort@data$t_alt_count/(lihc_cohort@data$t_ref_count+lihc_cohort@data$t_alt_count))*100
plotVaf(maf = lihc_cohort, vafCol = 'varian_allele_freq')


```


##Mutational status and survival

```{r}



lihc_cohort@clinical.data$survival_status <- ifelse(lihc_cohort@clinical.data$vital_status=="Alive", 1, 0)


mafSurvival(maf = lihc_cohort, genes = 'TP53', time = 'days_to_last_follow_up', Status = 'survival_status', isTCGA = TRUE)

#Using top 20 mutated genes to identify a set of genes (of size 2) to predict poor prognostic groups

prog_geneset = survGroup(maf = lihc_cohort, top = 20, geneSetSize = 2, time = "days_to_last_follow_up", Status = "survival_status", verbose = FALSE)
print(prog_geneset)

mafSurvGroup(maf = lihc_cohort, geneSet = c("TP53", "DNAH7"), time = "days_to_last_follow_up", Status = "survival_status")


```


```{r}


somaticInteractions(maf = lihc_cohort, top = 25, pvalue = c(0.05, 0.1))

lihc.sig = oncodrive(maf = lihc_cohort, AACol = 'HGVSp', minMut = 5, pvalMethod = 'zscore')

head(lihc.sig)

plotOncodrive(res = lihc.sig, fdrCutOff = 0.1, useFraction = TRUE, labelSize = 0.5)

lihc.pfam = pfamDomains(maf = lihc_cohort, AACol = 'HGVSp', top = 10)
#Protein summary (Printing first 7 columns for display convenience)
lihc.pfam$proteinSummary[,1:7, with = FALSE]
#Domain summary (Printing first 3 columns for display convenience)
lihc.pfam$domainSummary[,1:3, with = FALSE]



```

##Comparison of cohorts

```{r}
#oligodendrocytes
oligo = read.maf(maf = "data/cohortMAF.oligo.gz")
#GBM

GBM = read.maf(maf = "data/cohortMAF.GBM.gz")

oligo_vs_gbm <- mafCompare(m1 = oligo, m2 = GBM, m1Name = 'Oligodendroglioma', m2Name = 'GBM', minMut = 5)
print(oligo_vs_gbm)

forestPlot(mafCompareRes = oligo_vs_gbm, pVal = 0.01)

genes = c("IDH1", "PTEN", "NOTCH1", "CIC", "EGFR")
coOncoplot(m1 = oligo, m2 = GBM, m1Name = 'Oligodendroglioma', m2Name = 'GBM', genes = genes, removeNonMutated = TRUE)

coBarplot(m1 = oligo, m2 = GBM, m1Name = 'Oligodendroglioma', m2Name = 'GBM')

lollipopPlot2(m1 = oligo, m2 = GBM, m1_name = 'Oligodendroglioma', m2_name = 'GBM', gene = "EGFR", AACol1 = "HGVSp", AACol2 = "HGVSp")

```


```{r}


fab.ce = clinicalEnrichment(maf = lihc_cohort, clinicalFeature = 'gender')
fab.ce$groupwise_comparision[p_value < 0.05]


pws = pathways(maf = lihc_cohort, plotType = 'treemap')
#plotPathways(maf = lihc_cohort, pathlist = pws)
```

```{r}

#Requires BSgenome object
lich.tnm = trinucleotideMatrix(maf = lihc_cohort, prefix = '', add = TRUE, ref_genome = "BSgenome.Hsapiens.UCSC.hg38")

plotApobecDiff(tnm = lich.tnm, maf = lihc_cohort, pVal = 0.2)


#Run main function with maximum 6 signatures. 
library('NMF')
lihc.sign = estimateSignatures(mat = lich.tnm, nTry = 6, pConstant = 0.1, plotBestFitRes = FALSE, parallel = 2)

plotCophenetic(res = lihc.sign)

lihc.sig = extractSignatures(mat = lich.tnm, n = 3, pConstant = 0.1,  parallel = 2)

#Compare against original 30 signatures 
lihc.og30.cosm = compareSignatures(nmfRes = lihc.sig, sig_db = "legacy")
#Compare against updated version3 60 signatures 
llihc.v3.cosm = compareSignatures(nmfRes = lihc.sig, sig_db = "SBS")

maftools::plotSignatures(nmfRes = lihc.sig, title_size = 1.2, sig_db = "SBS")

```
